pkg/operator/ceph/object/user/controller.go

```diff
@@ -151,7 +171,46 @@ func (c *ObjectStoreUserController) createUser(context *clusterd.Context, u *cep
 		DisplayName: &displayName,
 	}
 	objContext := object.NewContext(context, u.Spec.Store, u.Namespace)
-
+	// Wait with exponential backoff for the objectstore to be created
+	err := wait.ExponentialBackoff(backoff, func() (ok bool, err error) {
+		// check if CephObjectStore CR is created
+		logger.Infof("waiting for CephObjectStore %s to be created", u.Spec.Store)
+		_, err = context.RookClientset.CephV1().CephObjectStores(u.Namespace).Get(u.Spec.Store, metav1.GetOptions{})
+		// wait only if objectStore is not found. return on any other error
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return false, nil
+			}
+			logger.Errorf("CephObjectStore %s could not be found. %+v", u.Spec.Store, err)
+			return true, err
+		}
+		// check if ObjectStore is initialized
+		// rook does this by starting the RGW pod(s)
+		selector := fmt.Sprintf("%s=%s,%s=%s",
+			"rgw", u.Spec.Store,
+			"app", "rook-ceph-rgw",
+		)
+		logger.Infof("waiting for CephObjectStore %s to be initialized", u.Spec.Store)
+		pods, err := context.Clientset.CoreV1().Pods(u.Namespace).List(metav1.ListOptions{LabelSelector: selector})
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return false, nil
+			}
+			logger.Errorf("CephObjectStore %s may not be initialized. %+v", u.Spec.Store, err)
+			return true, err
+		}
+		// check if atleast one pod is running
+		for _, pod := range pods.Items {
+			if pod.Status.Phase == v1.PodRunning {
```

Comment:
Why not use the `FieldSelector` in `ListOptions` to only list running pods instead of implementing this check manually?